x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/PlannerUtils.java

```diff
@@ -72,6 +81,35 @@ public static Tuple<PhysicalPlan, PhysicalPlan> breakPlanBetweenCoordinatorAndDa
         return new Tuple<>(coordinatorPlan, dataNodePlan.get());
     }
 
+    public static PhysicalPlan dataNodeReductionPlan(LogicalPlan plan, PhysicalPlan unused) {
+        var pipelineBreakers = plan.collectFirstChildren(Mapper::isPipelineBreaker);
+
+        if (pipelineBreakers.isEmpty() == false) {
+            UnaryPlan pipelineBreaker = (UnaryPlan) pipelineBreakers.get(0);
+            if (pipelineBreaker instanceof TopN topN) {
+                return new TopNExec(topN.source(), unused, topN.order(), topN.limit(), 0);
```

Comment:
The estimateRowSize value (here 0) can be improved. Couldn't find a quick enough way to make it work, though.

Reply:
A zero might make it try to emit very large pages. Which could cause CircuitBreaker exceptions. If you can't set it to something calculated I'd use a bigger number like 1500 or something.

Reply:
Yeah, I figured this would happen. Made the change in the last batch of updates. Will wait and see if the circuit breaker is triggered. Btw, it was the circuit breaker in previous runs that was tripped, not an actual oom, even if a heap dump was created.

Reply:
> Btw, it was the circuit breaker in previous runs that was tripped, not an actual oom, even if a heap dump was created.

Fwiw, this can happen if the test runs for too long (#104107).